### PR TITLE
Complete documentation coverage of registry operator

### PR DESCRIPTION
**Pull request by feloy**
**Please specify the area for this PR**

**What does does this PR do / why we need it**:

**Which issue(s) this PR fixes**:

Fixes https://github.com/devfile/api/issues/1015
Fixes partially https://github.com/devfile/api/issues/1233

**PR acceptance criteria**:

- [x] Test Coverage 
    - Are your changes sufficiently tested, and are any applicable test cases added or updated to cover your changes?
- [x] Gosec scans 
  - Fix all MEDIUM and higher findings and/or annotate a finding per gosec instructions: https://github.com/securego/gosec#annotating-code to address why a finding is not a security issue

Documentation
- [ ] Does the [registry operator documentation](https://github.com/devfile/devfile-web/tree/main/libs/docs/src/docs/no-version) need to updated with your changes?

**How to test changes / Special notes to the reviewer**:

### CODE REVIEW

File: DEVFILE_REGISTRY.md [hunk 1]
```diff
@@ -61,3 +61,171 @@ If you want to send telemetry information to your own Segment instance, specify
...
+metadata:
+  name: devfile-registry
+spec:
+  devfileIndex:
+    image: quay.io/devfile/devfile-index:next
+  telemetry:
+    registryName: test
+  storage:
+    enabled: true
+    ociRegistryImage: 2Gi
```
Comment (feloy) [inline DEVFILE_REGISTRY.md:158]: This should be `registryVolumeSize` IMO, but the code currently indicate this:

```
	RegistryVolumeSize string `json:"ociRegistryImage,omitempty"`

```

Do we want to change the code directly, or do we want to deprecate it?

File: DEVFILE_REGISTRY.md [hunk 2]
```diff
@@ -61,3 +61,171 @@ If you want to send telemetry information to your own Segment instance, specify
...
+metadata:
+  name: devfile-registry
+spec:
+  devfileIndex:
+    image: quay.io/devfile/devfile-index:next
+  telemetry:
+    registryName: test
+  tls:
+    enabled: true
+    ociRegistryImage: my-tls-secret
```
Comment (feloy) [inline DEVFILE_REGISTRY.md:205]: This should be `secretName` IMO, but the code currently indicates this:

```
	SecretName string `json:"ociRegistryImage,omitempty"`

```

Do we want to change the code directly, or do we want to deprecate it?

File: DEVFILE_REGISTRY.md [hunk 3]
```diff
@@ -61,3 +61,171 @@ If you want to send telemetry information to your own Segment instance, specify
...
+spec:
+  devfileIndex:
+    image: quay.io/devfile/devfile-index:next
+  telemetry:
+    registryName: test
+  headless: true
+EOF
+```
+
+## Using a Persistent Volume to store OCI images
```
Comment (johnmcollier) [inline DEVFILE_REGISTRY.md:137]: As part of https://github.com/devfile/api/issues/1093, we will be retiring the storage field for the devfile registry, as it's no longer used. So we should leave it out of the documentation

File: DEVFILE_REGISTRY.md (hunk 2 above)
Comment (johnmcollier) [reply DEVFILE_REGISTRY.md:205]: Preferably we should fix it before documenting it as it's a typo.

File: DEVFILE_REGISTRY.md (hunk 1 above)
Comment (johnmcollier) [reply DEVFILE_REGISTRY.md:158]: Preferably we should fix it before documenting it as it's a typo.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -18,9 +18,9 @@ You can tag Devfile Registry related issues with the `/area registry` text in yo
 
 #### First Time Setup
 1. Install prerequisites:
-   - Go 1.13 or higher
+   - Go 1.18 or higher
    - Docker or Podman
-   - Operator-SDK 1.11.0 or higher (including `controller-gen` 0.6.0 or higher)
+   - Operator-SDK 1.28.0 or higher (including `controller-gen` 0.10.0 or higher)
```
Comment (michael-valdron) [inline CONTRIBUTING.md:23]: We could link to [README Requirements](https://github.com/devfile/registry-operator/tree/main#requirements) instead or vice versa, as the content in these places are the same.

**Review by michael-valdron — COMMENTED** [review]
Looks great! I add one feedback comment on the first time setup. 

Additionally, I we'll need to run `make install-cert` before running `make install && make deploy`

**Comment by feloy** [comment]
> Looks great! I add one feedback comment on the first time setup.
> 
> Additionally, we'll need to run `make install-cert` before running `make install && make deploy`

Thanks for your review. I have pushed a new commit for first 2 points of https://github.com/devfile/api/issues/1233

I don't understand what the third one means (I cannot find such a command in Makefile):

```
 Include docker-buildx as an optional build and push step for multi arch
```

File: README.md [hunk 5]
```diff
@@ -71,34 +73,52 @@ Some of the rules supported by the makefile:
...
+| gosec | Check for security problems in non-test source files |
 
 To see all rules supported by the makefile, run `make help`
 
 ## Testing
 
 To run integration tests for the operator, run `make test-integration`. 
 
-By default, the tests will use the default image for the operator, `quay.io/devfile/registry-operator:next`. To use your own image, run:
+The `oc` executable must be accessible.
```
Comment (michael-valdron) [inline README.md:89]: In the current state this is required, however we will need to open an issue to allow `kubectl` to be used for the integration test suite given we support OpenShift and Kubernetes.

**Review by michael-valdron — APPROVED** [review]
/lgtm

> > Looks great! I add one feedback comment on the first time setup.
> > Additionally, we'll need to run `make install-cert` before running `make install && make deploy`
> 
> @michael-valdron Thanks for your review. I have pushed a new commit for first 2 points of [devfile/api#1233](https://github.com/devfile/api/issues/1233)
> 
> I don't understand what the third one means (I cannot find such a command in Makefile):
> 
> ```
>  Include docker-buildx as an optional build and push step for multi arch
> ```

This is a new rule that will be added in #45, I can include this doc change there.

**Pull request merged**